deps/v8/src/s390/macro-assembler-s390.cc

```diff
@@ -1324,8 +1324,8 @@ void MacroAssembler::CheckDebugHook(Register fun, Register new_target,
   ExternalReference debug_hook_active =
       ExternalReference::debug_hook_on_function_call_address(isolate());
   Move(r6, debug_hook_active);
-  tm(MemOperand(r6), Operand::Zero());
-  bne(&skip_hook);
+  tm(MemOperand(r6), Operand(0xFF));
+  beq(&skip_hook);
```

Comment:
Can you explain what the change in behavior is here?

`ExternalReference::debug_hook_on_function_call_address()` points to a `bool` so it's either 0 (false, disabled) or non-zero (true, enabled.)

Both the old and the new logic look wrong to me. This is what I would have expected:
```c++
tm(MemOperand(r6), Operand::Zero());
beq(&skip_hook);
```
I.e. `if (false == *(bool*)r6) goto skip_hook;`

Reply:
`tm` is test under mask. The 2nd operand is the mask. If mask is 0, according to ISA document, condition code will be 0x8 regardless what the 1st operand is. While in my actual solution, the 8 bit mask is all ones. TM will produce 0x4 if `Selected bits mixed zeros and ones`, 0x1 if `Selected bits all ones`. Either case falls under condition `(false == *(bool*)r6)`. `bne` uses mask 0x7, which covers both case.

Reply:
Ah, I understand now. Thanks for the explanation!